lib/searchable_model.rb

```diff
@@ -38,28 +38,29 @@ def search(search, page, user, includes={})
     # pass in a username to limit the search to the users items
     if (!user.nil?) && (!user.id.nil?)
       if column_names.include? 'user_id'
-        if self == User
-          sql_conditions = ""
-        else
+        if self != User
           # sql_conditions = "(#{table_name}.user_id = ? or #{table_name}.public = '1') and "
           sql_conditions = "(#{table_name}.user_id = ?) and "
           sql_parameters << user.id
         end
       end
     end
 
-    # debugger
-    sql_conditions = sql_conditions + '(' + searchable_attributes.collect {|a| "#{table_name}.#{a} like ?"}.join(' or ') + ')'
-    # FIXME - This search should do the following: split the terms based on whitespace, then perform the search
-    # like this.  Given fields a, b, and c, and terms x and y we want to query in this way:
-    # (a like x OR b like x OR c like x) AND (a like y OR b like y OR c like y)
-    # we should also allow the search not to break on whitespace enclosed in quotes... maybe tokenize it?
-    # iterate over the string, look for quote or just start the strings into an array
-    # we could first split then check each part and see if it has a quote or not.  If one has a quote, it needs to be combined with the subsequent one until the other quote is found
-    # do a search to see if we have quoted strings, and then replace the whitespace in the quoted strings with another character, then do the split, then fix the whitespace
-    # or just only allow quotes at start and end
-    # maybe we'll need to add a nested block.
-    searchable_attributes.length.times {sql_parameters << "%#{search}%"}
+    if !search.nil? && !search.empty?
+      search_terms = get_search_terms(search)
+
+      search_terms.each do |st|
+        sql_conditions = sql_conditions + ' (' + searchable_attributes.collect {|a| "#{table_name}.#{a} like ?"}.join(' or ') + ')'
+        if !st.equal?(search_terms.last)
+          sql_conditions = sql_conditions + ' and'
+        end
```

Comment:
I'd suggest changing this to:
```
new_sql_conditions =  search_terms.map do
  ' (' + searchable_attributes.collect {|a| "#{table_name}.#{a} like ?"}.join(' or ') + ')'
end
sql_conditions = sql_conditions + new_sql_conditions.join(' and')
```
It is a little more concise, and since the `st` is no longer needed it reduces the chance that a search term ends up directly in the sql instead of as a parameter.

Reply:
OK, thanks. I've made this change. I also tested out the access rules, and it looks to me like they're still working correctly.